core/wasmWasi/src/internal/Platform.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019-2024 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.internal
+
+import kotlinx.datetime.Instant
+import kotlin.wasm.WasmImport
+import kotlin.wasm.unsafe.Pointer
+import kotlin.wasm.unsafe.UnsafeWasmMemoryApi
+import kotlin.wasm.unsafe.withScopedMemoryAllocator
+
+/**
+ * Return the time value of a clock. Note: This is similar to `clock_gettime` in POSIX.
+ */
+@WasmImport("wasi_snapshot_preview1", "clock_time_get")
+private external fun wasiRawClockTimeGet(clockId: Int, precision: Long, resultPtr: Int): Int
+
+private const val CLOCKID_REALTIME = 0
+
+@OptIn(UnsafeWasmMemoryApi::class)
+private fun clockTimeGet(): Long = withScopedMemoryAllocator { allocator ->
+    val rp0 = allocator.allocate(8)
+    val ret = wasiRawClockTimeGet(
+        clockId = CLOCKID_REALTIME,
+        precision = 1,
+        resultPtr = rp0.address.toInt()
+    )
+    if (ret == 0) {
+        (Pointer(rp0.address.toInt().toUInt())).loadLong()
+    } else {
+        error("WASI call failed with $ret")
+    }
+}
+
+internal actual fun currentTime(): Instant = clockTimeGet().let { time ->
+    Instant(time / NANOS_PER_ONE, (time % NANOS_PER_ONE).toInt())
+}
+
+internal actual fun currentSystemDefaultZone(): Pair<String, TimeZoneRules?> =
+    throw UnsupportedOperationException("WASI platform does not support system timezone obtaining")
```

Comment:
You would like to add it in common `TimeZone` KDoc or in make a separate KDoc for platform-specific  `currentSystemDefaultZone`?

Reply:
The common one, because when writing `TimeZone.currentSystemDefault()` in common code, you should still be aware that this may fail.

Reply:
After an internal discussion, we decided that it makes more sense to return the UTC time zone by default. The reason is, because `TimeZone.currentSystemDefault()` is not easily mockable (see https://github.com/Kotlin/kotlinx-datetime/issues/17), sometimes, there are calls to `currentSystemDefault()` deep inside the existing code, and to have them throw exceptions now would be problematic. If/when we replace `currentSystemDefault()` with a version that supports dependency injection better, that version will throw on WASI.

This should still be documented, and it should be in the common code.

Reply:
Yes.